### DISCUSSION TITLE
Required System Resources

### DISCUSSION THREAD

**Discussion by infinitydon** [HewlettPackard/PacketRusher#44] [Q&A] [answered]
Hi,

Can you kindly provide the system resources especially like number of CPU cores and RAM that you used to achieve the following:

```
-- We roughly reach 5 GB/s per UE, which is more than what a real UE can achieve.

-- Simulate multiple UEs and gNodeB from a single tool
        We tested up to 10k UEs!
```

**Comment by linouxis9** [comment, marked as answer]
Hi @infinitydon,

The VM where our tests were conducted had 8 vCPU and 24 GB of RAM.
What allowed us to reach this kind of speed was two fold:
- The N2/N3 network interface used by PacketRusher was not a virtual device, but a Broadcom 25G network card being PCI-E passthrough-ed to the virtual machine.
- We used a proprietary UPF with another dedicated 25G network card, and both the UPF's VM and PacketRusher's VM connected through a dedicated 25G switch. You could probably use a DAC cable between the two interfaces and avoid paying for an expensive switch.

No open source UPF I've tested has reached this throughput, especially not completely in the cloud without some kind of bare metal access. [UPG-VPP](https://github.com/travelping/upg-vpp) might be your best option with DPDK and a real network device.
You may take a look at the speed measurements s5ushida was able to take with PacketRusher and the open source UPFs in far from the best conditions (without PCI-E Passthrough for either PacketRusher or the UPF), but it sets a nice floor and shows very interesting [results](https://github.com/s5uishida/simple_measurement_of_upf_performance)!

If you have any more questions, feel free to reach out to me.

Thanks and cheers,
Valentin

**Reply by infinitydon** [reply]
Thanks for the response
